registration/admin.py

```diff
@@ -58,7 +58,9 @@ def save_model(self, request, obj, form, change):
         """
         update_fields = []
         if form and form.changed_data:
-            if form.initial['is_staff'] != form.cleaned_data['is_staff']:
+            initial = form.initial.get('is_staff')
+            new = form.cleaned_data.get('is_staff')
+            if not (initial is None or new is None) and initial != new:
```

Comment:
```suggestion
            # Test for None to update only if it was sent in the form.
            if new is not None and initial != new:
```
It would suspect, that only the new value needs to be different from `None`, to trigger an update.